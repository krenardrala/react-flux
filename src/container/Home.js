import React, { Component } from 'react'
import {
  Responsive
} from 'semantic-ui-react'

import MainHeader from '../component/Header';
import HomepageIntro from '../component/HomepageIntro';
import HomepageGrid from '../component/HomepageGrid';
import Footer from '../component/Footer';

class HomepageLayout extends Component {
  render() {
    return (
      <Responsive>
        <HomepageIntro/>
        <HomepageGrid/>
        <Footer/>
      </Responsive>
    );
  }
}

export default HomepageLayout