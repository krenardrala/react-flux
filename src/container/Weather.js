import React, { Component } from 'react'
import {
  Responsive,
  Grid,
  Input,
  Button,
  Container
} from 'semantic-ui-react';
import * as WeatherActions from '../actions/WeatherActions';
import MainHeader from '../component/Header';
import Footer from '../component/Footer';
import SearchCity from '../component/SearchCity';
import WeatherStore from '../stores/WeatherStore';

class Weather extends Component {

  state = {
    cityName: '',
    weather: []
  }

  handleSearchCity = () => {
    let city = this.state.cityName;
    WeatherActions.getWeather(city);
    /*this.setState({
      weather: WeatherStore.getAll()
    })*/
  }

  handleInputChange = (e) => {
    this.setState({[e.target.name]: e.target.value });
  }

  componentDidMount() {
    WeatherStore.on("change", () => {
      this.setState({
        weather: WeatherStore.getAll()
      });
    })
  }

  componentWillUnmount() {
    WeatherStore.removeListener("change", () => {
      this.setState({
        weather: WeatherStore.getAll()
      });
    });
  }

  render() {
    return(
      <Responsive>
        <MainHeader/>
        <Container fluid style={{minHeight: '650px', paddingTop: '100px'}}>
          <Grid columns={2} padded='vertically'>
            <Grid.Column textAlign="center">
              <Input size='big' icon='search' placeholder='Search...' onChange={this.handleInputChange} name="cityName" />
              <Button size='big' color="teal" style={{marginLeft: '15px'}} onClick={this.handleSearchCity}>Search</Button>
            </Grid.Column>
            <Grid.Column>
              <SearchCity weather={this.state.weather}/>
            </Grid.Column>
          </Grid>
        </Container>
        <Footer/>
      </Responsive>
    );
  }
}

export default Weather