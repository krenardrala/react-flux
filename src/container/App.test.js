import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Home from './Home';
import Weather from './Weather';

describe('Home', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Home debug />);

    expect(component).toMatchSnapshot();
  });
});

describe('Weather', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Weather debug />);

    expect(component).toMatchSnapshot();
  });
});

it('should render correctly with no props', () => {
  const component = shallow(<Home/>);

  expect(component).toMatchSnapshot();
});

it('should render correctly with no props', () => {
  const component = shallow(<Weather/>);

  expect(component).toMatchSnapshot();
});

it('should be possible to activate button with Spacebar', () => {
  const component = mount(<Weather />);
  component
    .find('button')
    .simulate('keydown', { keyCode: 32 });
  expect(component).toMatchSnapshot();
  component.unmount();
});
