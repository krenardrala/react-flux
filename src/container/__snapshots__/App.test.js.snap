// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly in "debug" mode 1`] = `
<Responsive
  getWidth={[Function]}
>
  <HomepageIntro />
  <HomepageIntro />
  <Footer />
</Responsive>
`;

exports[`Weather should render correctly in "debug" mode 1`] = `
<Responsive
  getWidth={[Function]}
>
  <Container
    fluid={true}
    style={
      Object {
        "minHeight": "650px",
        "paddingTop": "100px",
      }
    }
  >
    <Grid
      columns={2}
      padded="vertically"
    >
      <GridColumn
        textAlign="center"
      >
        <Input
          icon="search"
          name="cityName"
          onChange={[Function]}
          placeholder="Search..."
          size="big"
          type="text"
        />
        <Button
          as="button"
          color="teal"
          onClick={[Function]}
          size="big"
          style={
            Object {
              "marginLeft": "15px",
            }
          }
        >
          Search
        </Button>
      </GridColumn>
      <GridColumn>
        <SearchCity
          weather={Array []}
        />
      </GridColumn>
    </Grid>
  </Container>
  <Footer />
</Responsive>
`;

exports[`should be possible to activate button with Spacebar 1`] = `
<Weather>
  <Responsive
    getWidth={[Function]}
  >
    <div>
      <Container
        fluid={true}
        style={
          Object {
            "minHeight": "650px",
            "paddingTop": "100px",
          }
        }
      >
        <div
          className="ui fluid container"
          style={
            Object {
              "minHeight": "650px",
              "paddingTop": "100px",
            }
          }
        >
          <Grid
            columns={2}
            padded="vertically"
          >
            <div
              className="ui vertically padded two column grid"
            >
              <GridColumn
                textAlign="center"
              >
                <div
                  className="center aligned column"
                >
                  <Input
                    icon="search"
                    name="cityName"
                    onChange={[Function]}
                    placeholder="Search..."
                    size="big"
                    type="text"
                  >
                    <div
                      className="ui big icon input"
                    >
                      <input
                        name="cityName"
                        onChange={[Function]}
                        placeholder="Search..."
                        type="text"
                      />
                      <Icon
                        as="i"
                        name="search"
                      >
                        <i
                          aria-hidden="true"
                          className="search icon"
                          onClick={[Function]}
                        />
                      </Icon>
                    </div>
                  </Input>
                  <Button
                    as="button"
                    color="teal"
                    onClick={[Function]}
                    size="big"
                    style={
                      Object {
                        "marginLeft": "15px",
                      }
                    }
                  >
                    <button
                      className="ui teal big button"
                      onClick={[Function]}
                      style={
                        Object {
                          "marginLeft": "15px",
                        }
                      }
                    >
                      Search
                    </button>
                  </Button>
                </div>
              </GridColumn>
              <GridColumn>
                <div
                  className="column"
                >
                  <SearchCity
                    weather={Array []}
                  />
                </div>
              </GridColumn>
            </div>
          </Grid>
        </div>
      </Container>
      <Footer>
        <Segment
          inverted={true}
          style={
            Object {
              "padding": "5em 0em",
            }
          }
          vertical={true}
        >
          <div
            className="ui inverted vertical segment"
            style={
              Object {
                "padding": "5em 0em",
              }
            }
          >
            <Container>
              <div
                className="ui container"
              >
                <Grid
                  divided={true}
                  inverted={true}
                  stackable={true}
                >
                  <div
                    className="ui inverted stackable divided grid"
                  >
                    <GridRow>
                      <div
                        className="row"
                      >
                        <GridColumn
                          width={3}
                        >
                          <div
                            className="three wide column"
                          >
                            <Header
                              as="h4"
                              content="About"
                              inverted={true}
                            >
                              <h4
                                className="ui inverted header"
                              >
                                About
                              </h4>
                            </Header>
                            <List
                              inverted={true}
                              link={true}
                            >
                              <div
                                className="ui inverted link list"
                                role="list"
                              >
                                <ListItem
                                  as="a"
                                >
                                  <a
                                    className="item"
                                    onClick={[Function]}
                                    role="listitem"
                                  >
                                    Sitemap
                                  </a>
                                </ListItem>
                                <ListItem
                                  as="a"
                                >
                                  <a
                                    className="item"
                                    onClick={[Function]}
                                    role="listitem"
                                  >
                                    Contact Us
                                  </a>
                                </ListItem>
                                <ListItem
                                  as="a"
                                >
                                  <a
                                    className="item"
                                    onClick={[Function]}
                                    role="listitem"
                                  >
                                    Religious Ceremonies
                                  </a>
                                </ListItem>
                                <ListItem
                                  as="a"
                                >
                                  <a
                                    className="item"
                                    onClick={[Function]}
                                    role="listitem"
                                  >
                                    Gazebo Plans
                                  </a>
                                </ListItem>
                              </div>
                            </List>
                          </div>
                        </GridColumn>
                        <GridColumn
                          width={3}
                        >
                          <div
                            className="three wide column"
                          >
                            <Header
                              as="h4"
                              content="Services"
                              inverted={true}
                            >
                              <h4
                                className="ui inverted header"
                              >
                                Services
                              </h4>
                            </Header>
                            <List
                              inverted={true}
                              link={true}
                            >
                              <div
                                className="ui inverted link list"
                                role="list"
                              >
                                <ListItem
                                  as="a"
                                >
                                  <a
                                    className="item"
                                    onClick={[Function]}
                                    role="listitem"
                                  >
                                    Banana Pre-Order
                                  </a>
                                </ListItem>
                                <ListItem
                                  as="a"
                                >
                                  <a
                                    className="item"
                                    onClick={[Function]}
                                    role="listitem"
                                  >
                                    DNA FAQ
                                  </a>
                                </ListItem>
                                <ListItem
                                  as="a"
                                >
                                  <a
                                    className="item"
                                    onClick={[Function]}
                                    role="listitem"
                                  >
                                    How To Access
                                  </a>
                                </ListItem>
                                <ListItem
                                  as="a"
                                >
                                  <a
                                    className="item"
                                    onClick={[Function]}
                                    role="listitem"
                                  >
                                    Favorite X-Men
                                  </a>
                                </ListItem>
                              </div>
                            </List>
                          </div>
                        </GridColumn>
                        <GridColumn
                          width={7}
                        >
                          <div
                            className="seven wide column"
                          >
                            <Header
                              as="h4"
                              inverted={true}
                            >
                              <h4
                                className="ui inverted header"
                              >
                                Footer Header
                              </h4>
                            </Header>
                            <p>
                              Extra space for a call to action inside the footer that could help re-engage users.
                            </p>
                          </div>
                        </GridColumn>
                      </div>
                    </GridRow>
                  </div>
                </Grid>
              </div>
            </Container>
          </div>
        </Segment>
      </Footer>
    </div>
  </Responsive>
</Weather>
`;

exports[`should render correctly with no props 1`] = `
<Responsive
  getWidth={[Function]}
>
  <HomepageIntro />
  <HomepageIntro />
  <Footer />
</Responsive>
`;

exports[`should render correctly with no props 2`] = `
<Responsive
  getWidth={[Function]}
>
  <Container
    fluid={true}
    style={
      Object {
        "minHeight": "650px",
        "paddingTop": "100px",
      }
    }
  >
    <Grid
      columns={2}
      padded="vertically"
    >
      <GridColumn
        textAlign="center"
      >
        <Input
          icon="search"
          name="cityName"
          onChange={[Function]}
          placeholder="Search..."
          size="big"
          type="text"
        />
        <Button
          as="button"
          color="teal"
          onClick={[Function]}
          size="big"
          style={
            Object {
              "marginLeft": "15px",
            }
          }
        >
          Search
        </Button>
      </GridColumn>
      <GridColumn>
        <SearchCity
          weather={Array []}
        />
      </GridColumn>
    </Grid>
  </Container>
  <Footer />
</Responsive>
`;
